### PR TITLE
refactor(sub): move crypto functions into their own module

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

File: mergify_engine/tests/unit/test_crypto.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+from mergify_engine import crypto
+
+
+def test_encrypt():
+    x = "this is an amazing string, right? 🙄".encode()
+    assert crypto.decrypt(crypto.encrypt(x))
```
Comment (sileht) [inline mergify_engine/tests/unit/test_crypto.py:21]: maybe we need also `== x` ?

File: mergify_engine/tests/unit/test_crypto.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    assert crypto.decrypt(crypto.encrypt(x))
```
Comment (sileht) [inline mergify_engine/tests/unit/test_crypto.py:22]: Duplicate line

File: mergify_engine/tests/unit/test_crypto.py (hunk 1 above)
Comment (jd) [reply mergify_engine/tests/unit/test_crypto.py]: 🤦

File: mergify_engine/tests/unit/test_crypto.py (hunk 2 above)
Comment (jd) [reply mergify_engine/tests/unit/test_crypto.py]: This was actually voluntary to test there was no side effect on the crypto module itself.

**Review by sileht — APPROVED** [review]

**Pull request merged**